includes/class-activitypub.php

```diff
@@ -115,9 +116,37 @@ public static function render_json_template( $template ) {
 			$json_template = ACTIVITYPUB_PLUGIN_DIR . '/templates/blog-json.php';
 		}
 
+		/*
+		 * Check if the request is authorized.
+		 *
+		 * @see https://www.w3.org/wiki/SocialCG/ActivityPub/Primer/Authentication_Authorization#Authorized_fetch
+		 * @see https://swicg.github.io/activitypub-http-signature/#authorized-fetch
+		 */
 		if ( ACTIVITYPUB_AUTHORIZED_FETCH ) {
 			$verification = Signature::verify_http_signature( $_SERVER );
 			if ( \is_wp_error( $verification ) ) {
+				header( 'HTTP/1.1 401 Unauthorized' );
+
+				add_filter(
+					'activitypub_activity_object_array',
+					function ( $activity ) {
+						$activity = array_intersect_key(
+							$activity,
+							array_flip(
+								array(
+									'@context',
+									'id',
+									'type',
+									'preferredUsername',
+									'publicKey',
+								)
+							)
+						);
+
+						return $activity;
+					}
+				);
+
 				// fallback as template_loader can't return http headers
 				return $template;
```

Comment:
Only requests for the Application actor should be allowed to bypass signature verification (to prevent a deadlock loop).

The current code sends a 401 and the whole actor object!

Reply:
Also, I'm not sure the application actor needs a non-REST profile!?

So the above mentioned signature-bypass should probably only apply in the REST Server class.

(Unless I am missing something :-)

Reply:
The whole object? It should only send a limited subset with the signature!?! Can you verify this? But I can remove it completely if the Application is enough.

Reply:
Right, not the whole actor, but this would prevent enforcing a server level block.